oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -1393,7 +1393,7 @@ private void updateCoords(String cluster, BundleJob bundle,
 
             //calculate next start time based on delay.
             endTime = (delay == null) ? endTime
-                    : EntityUtil.getNextStartTime(coord.getStartTime(), delay, EntityUtil.getTimeZone(entity), endTime);
+                    : EntityUtil.getNextInstanceTimeWithDelay(endTime, delay, EntityUtil.getTimeZone(entity));
```

Comment:
This change basically makes sure that the delay is added to endTime (clusterDef.getValidity().getEnd()). Can you please explain how it solves for the holes, in between, during update?